checker/lib/README.developers

```diff
@@ -0,0 +1,11 @@
+Please read the README file in this directory first, as it is the file
+that is shipped in the distribution and is intended for users of the
+Checker Framework.
+
+When adding a .jar file to this directory, update the list in README and also
+the list in ../manual/annotating-libraries.tex .
+In general, old files need not be removed; that way, we accommodate users
+regardless of which version of a library they wish to use.
+
+To change which files from this directory are included in the distribution,
```

Comment:
I find it confusing that there is both a checker-includes and a checker-excludes file, and that neither one is documented anywhere in the repository.  What is the relationship between them?  Why do we need both of them?  Why is checker-includes so large?

We should strive to minimize the differences between the repository and the release.  Therefore, anything that appears in either checker-includes or checker-excludes is suspect.  I propose that we attempt to make those files as small as possible, and that we add a comment (prefixed by a # sign permitting tools to strip those lines off before interpreting the rest) to every line to justify it.